NYTPhotoViewerTests/NYTPhotoViewerSinglePhotoDataSourceTests.m

```diff
@@ -18,6 +18,10 @@ @interface NYTPhotoViewerSinglePhotoDataSourceTests : XCTestCase
 
 @implementation NYTPhotoViewerSinglePhotoDataSourceTests
 
+// The next 3 tests _deliberately_ pass nil, we don't need to be warned about it.
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wnonnull"
```

Comment:
If clang was complaining about passing nil, does that mean the method is annotated wrong? If we're testing whether we can pass nil, doesn't that mean we support it?

Reply:
I think the tests were written because ObjC allows you pass nil, and if you do, we hope not to throw an exception.